--- a/include/configs/mx6_cubox-i.h
+++ b/include/configs/mx6_cubox-i.h
@@ -239,7 +239,9 @@
                         "setenv fdt_file ${fdt_prefix}-hummingboard${somrev}.dtb; " \
                 "else " \
                         "setenv fdt_file ${fdt_prefix}-hummingboard2${somrev}.dtb; " \
-                "fi;\0" \
+                "fi; " \
+                "setenv fdtfile ${fdt_file};\0" \
+        "findfdt=run autodetectfdt;\0" \
         "loadbootenv=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${file_prefix}${bootenv};\0" \
         "loadfdt=if test ${boottype} = mmc; then " \
 				"for prefix in ${fdt_prefixes}; do " \
