--- a/include/configs/mx6_cubox-i.h
+++ b/include/configs/mx6_cubox-i.h
@@ -186,8 +186,10 @@
         "splashpos=m,m\0" \
         "fdt_high=0xffffffff\0" \
         "initrd_high=0xffffffff\0" \
-        "fdt_addr=0x18000000\0" \
+        "fdt_addr=0x14700000\0" \
         "boot_fdt=try\0" \
+        "ramdisk_addr=0x14800000\0" \
+        "ramdisk_file=initrd\0" \
         "ip_dyn=yes\0" \
         "mmcdev=" __stringify(CONFIG_SYS_MMC_ENV_DEV) "\0" \
         "mmcpart=1\0" \
@@ -249,6 +251,7 @@
                 "if test -n ${ramdisk_file}; then " \
 		    "if run loadramdisk; then " \
 			"echo Loaded ${ramdisk_file}; " \
+                        "setenv ramdisk_addr ${ramdisk_addr}:0x${filesize}; " \
 		    "else " \
 			"setenv ramdisk_addr -; " \
 		    "fi; " \
@@ -387,4 +390,7 @@
 #define CONFIG_CMD_CACHE
 #endif
 
+/* enable raw initrd`s (that is, initrd not processed by mkimage) */
+#define CONFIG_SUPPORT_RAW_INITRD
+
 #endif			       /* __CONFIG_H * */
